server/controllers/applicant/getApplicantsStats.js

```diff
@@ -0,0 +1,36 @@
+const Applicant = require('../../database/models/applicant');
+
+const getApplicantsStats = (req, res, next) => {
+  Applicant.find({}, (err, applicants) => {
+    if (err) {
+      return res.status(400).json({ statusCode: 400, error: 'Bad Request' });
+    }
+    if (!applicants.length) {
+      return res.status(404).json({ statusCode: 404, error: `Data not found` });
+    }
+    return true;
+  });
```

Comment:
@alidah0 I think this Applicant.find() block (line 4-12) isn't providing that much value and could be removed, for a couple of reasons:
- find() and its callback are async, so it could theoretically finish *after* the Promise.all code below
- In my opinion, having no applicants (applicants.length) is not an error condition or representative of something "not found" - I think it's acceptable to just return a 200 response with the stats all equal to zero (which would happen automatically with your code below)
- The other error handler, line 6, would probably also be handled properly by the catch() on line 33

Reply:
@tomduggan85  oh! I didn't notice that zeros still stats. Thank you so much for your notes, I will fix all these but do you think promise.all is a good way to execute multiple queries at once.  what do think?

Reply:
Yes, promise.all is great for that!